iohk/NixOps.hs

```diff
@@ -0,0 +1,531 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE ViewPatterns #-}
+
+module NixOps where
+
+import Prelude hiding (FilePath)
+import           Control.Exception (throwIO)
+import qualified Data.Aeson                    as AE
+import           Data.Aeson                       ((.:), (.=))
+import qualified Data.ByteString.Lazy          as BL
+import qualified Data.ByteString.UTF8          as BUTF8
+import Data.Char (ord)
+import qualified Data.Yaml                     as YAML
+import Data.Yaml (FromJSON(..), ToJSON(..))
+import           Data.Maybe
+import qualified Data.Map                      as Map
+import Data.Monoid ((<>))
+import           Data.Optional (Optional)
+import           Data.List                        (sort)
+import qualified Data.Set                      as Set
+import qualified Data.Text as T
+import qualified Data.Text.IO as TIO
+import Data.Text.Lazy (fromStrict)
+import Data.Text.Lazy.Encoding (encodeUtf8)
+import Data.Csv (decodeWith, FromRecord(..), FromField(..), HasHeader(..), defaultDecodeOptions, decDelimiter)
+import qualified Data.Vector as V
+import Data.ByteString.Lazy.Char8 (ByteString, pack)
+import qualified Filesystem.Path.CurrentOS     as Path
+import GHC.Generics
+import Safe (headMay)
+import           Text.Read                        (readMaybe)
+import Turtle hiding (procs)
+import qualified Turtle as Turtle
+
+
+-- * Constants
+--
+iohkNixopsURL, cardanoSlURL, cardanoSlExplorerURL, awsPublicIPURL :: URL
```

Comment:
The purpose of `NixOps.hs` has been to have functions abstracted over NixOps API. Now it mixes a lot of cardano-sl specific things, losing the split between the functions in `iohk-ops.hs` and `NixOps.hs`